### PR TITLE
Initial release of reputation UI

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Both

* **Description of Pull Request**: 
Adds support for the new reputation UI.

### CODE REVIEW

**Comment by gravity-ro** [comment]
No monsters to this function.

Screenshot_3

File: db/import-tmpl/reputation.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+###########################################################################
+#
+# Reputation Settings
+#
+###########################################################################
+# - Id                                  Clientside index
+#   Name                                Name of the reputation type
+#   Variable                            Name of the variable that will store the reputation (Default: RepPoints<id>)
+#   Minimum                             Minimum amount of points (Default: INT64_MIN)
+#   Maximum                             Maximum amount of points (Default: INT64_MAX)
```
Comment (aleos89) [inline db/import-tmpl/reputation.yml:29]:
```suggestion
# - Id                                  Client side index.
#   Name                                Name of the reputation type.
#   Variable                            Name of the variable that will store the reputation. (Default: RepPoints<id>)
#   Minimum                             Minimum amount of points. (Default: INT64_MIN)
#   Maximum                             Maximum amount of points. (Default: INT64_MAX)
```

File: db/re/reputation.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+###########################################################################
+#
+# Reputation Settings
+#
+###########################################################################
+# - Id                                  Clientside index
+#   Name                                Name of the reputation type
+#   Variable                            Name of the variable that will store the reputation
+#   Minimum                             Minimum amount of points (Default: INT64_MIN)
+#   Maximum                             Maximum amount of points (Default: INT64_MAX)
```
Comment (aleos89) [inline db/re/reputation.yml:29]:
```suggestion
# - Id                                  Client side index.
#   Name                                Name of the reputation type.
#   Variable                            Name of the variable that will store the reputation. (Default: RepPoints<id>)
#   Minimum                             Minimum amount of points. (Default: INT64_MIN)
#   Maximum                             Maximum amount of points. (Default: INT64_MAX)
```

File: db/reputation.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+###########################################################################
+#
+# Reputation Settings
+#
+###########################################################################
+# - Id                                  Clientside index
+#   Name                                Name of the reputation type
+#   Variable                            Name of the variable that will store the reputation (Default: RepPoints<id>)
+#   Minimum                             Minimum amount of points (Default: INT64_MIN)
+#   Maximum                             Maximum amount of points (Default: INT64_MAX)
```
Comment (aleos89) [inline db/reputation.yml:29]:
```suggestion
# - Id                                  Client side index.
#   Name                                Name of the reputation type.
#   Variable                            Name of the variable that will store the reputation. (Default: RepPoints<id>)
#   Minimum                             Minimum amount of points. (Default: INT64_MIN)
#   Maximum                             Maximum amount of points. (Default: INT64_MAX)
```

File: doc/script_commands.txt [hunk 4]
```diff
@@ -8174,6 +8174,18 @@ Opens the enchantgrade UI for the attached character or the player given by the
 
 This command requires packet version 2020-07-24 or newer.
 
+---------------------------------------
+
+*set_reputation_points(<type>,<points>{,<char id>})
+
+Sets the reputation points <points> for reputation group <type> for the attached player or the given character id.
```
Comment (aleos89) [inline doc/script_commands.txt:8181]:
```suggestion
Sets the reputation points via <points> for reputation group <type> for the attached player or the given character ID.
```

File: doc/script_commands.txt [hunk 5]
```diff
@@ -8174,6 +8174,18 @@ Opens the enchantgrade UI for the attached character or the player given by the
...
+
+---------------------------------------
+
+*get_reputation_points(<type>{,<char id>})
+
+Gets the reputation points for reputation group <type> for the attached player or the given character id.
```
Comment (aleos89) [inline doc/script_commands.txt:8187]:
```suggestion
Gets the reputation points for reputation group <type> for the attached player or the given character ID.
```

File: src/map/script.cpp [hunk 6]
```diff
@@ -26038,6 +26038,60 @@ BUILDIN_FUNC( enchantgradeui ){
...
 
+BUILDIN_FUNC(set_reputation_points){
+	struct map_session_data* sd;
+
+	if( !script_charid2sd( 4, sd ) ){
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int64 type = script_getnum64( st, 2 );
+
```
Comment (aleos89) [inline src/map/script.cpp:26049]:
```suggestion
```

File: src/map/script.cpp [hunk 7]
```diff
@@ -26038,6 +26038,60 @@ BUILDIN_FUNC( enchantgradeui ){
...
+BUILDIN_FUNC(get_reputation_points){
+	struct map_session_data* sd;
+
+	if( !script_charid2sd( 3, sd ) ){
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int64 type = script_getnum64( st, 2 );
+
```
Comment (aleos89) [inline src/map/script.cpp:26078]:
```suggestion
```

**Review by aleos89 — APPROVED** [review]
LGTM!

**Review by null6385 — APPROVED** [review]

File: doc/script_commands.txt (hunk 4 above)
Comment (Atemo) [reply doc/script_commands.txt:8181]: Can you add a sentence to say that the `<type>` is the "Client side index"? the expected parameter is not clear enough

**Review by null6385 — APPROVED** [review]

**Review by Atemo — APPROVED** [review]

**Comment by dimasshotta** [comment]
is the script command set_reputation_points supposed to be set the reputation point? I thought that it supposed to be adding the points, not only set.. because,  in KRO you will gain the point everytime you finish a quest. seems set_reputation_point is not adding the point since it doesn't look to the current character's reputation point.

**Comment by Escanor-10** [comment]
When I use set_reputation_points(3),100; and in another npc I use set_reputation_points(3),50; reputation does not add up, could you help me, please.

**Pull request merged**